### PR TITLE
Adds a binary that runs over trajectory and data to create assets of all point data.

### PR DESCRIPTION
**Pull request by SirVer**

### CODE REVIEW

File: cartographer_ros/cartographer_ros/assets_writer_main.cc [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+              "First directory in which configuration files are searched, "
+              "second is always the Cartographer installation to allow "
+              "including files from there.");
+DEFINE_string(configuration_basename, "",
+              "Basename, i.e. not containing any directory prefix, of the "
+              "configuration file.");
+DEFINE_string(
+    urdf, "",
+    "URDF file that contains static links for your sensor configuration.");
+DEFINE_string(bag_filename, "", "The bag to run over.");
```
Comment (damonkohler) [inline cartographer_ros/cartographer_ros/assets_writer_main.cc]: Bag to process.

File: cartographer_ros/cartographer_ros/assets_writer_main.cc [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+DEFINE_string(trajectory_filename, "", "The proto containing the trajectory.
```
Comment (damonkohler) [inline cartographer_ros/cartographer_ros/assets_writer_main.cc]: Proto containing the trajectory written by the /finish_trajectory service.

File: cartographer_ros/cartographer_ros/assets_writer_main.cc [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+              Written by the node on /finish_trajectory.");
+
+namespace cartographer_ros {
+  namespace {
```
Comment (damonkohler) [inline cartographer_ros/cartographer_ros/assets_writer_main.cc]: Formatting?

File: cartographer_ros/cartographer_ros/assets_writer_main.cc [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+  ReadTrajectory(const std::string& trajectory_filename) {
+    std::ifstream stream(trajectory_filename.c_str());
+    carto::proto::Trajectory trajectory_proto;
+    CHECK(trajectory_proto.ParseFromIstream(&stream));
+    return carto::transform::TransformInterpolationBuffer::FromTrajectory(
+        trajectory_proto);
+  }
+
+  void ReadStaticTransformsFromUrdf(const string& urdf_filename,
+                                    ::tf2_ros::Buffer* buffer) {
```
Comment (damonkohler) [inline cartographer_ros/cartographer_ros/assets_writer_main.cc]: const

File: cartographer_ros/cartographer_ros/assets_writer_main.cc [hunk 5]
```diff
@@ -0,0 +1,191 @@
...
+      transform.header.frame_id = link->getParent()->name;
+      buffer->setTransform(transform, "urdf", true /* is_static */);
+    }
+  }
+
+  void Run(const string& trajectory_filename, const string& bag_filename,
+           const string& configuration_directory,
+           const string& configuration_basename, const string& urdf_filename) {
+    ::tf2_ros::Buffer buffer;
+    ReadStaticTransformsFromUrdf(urdf_filename, &buffer);
```
Comment (damonkohler) [inline cartographer_ros/cartographer_ros/assets_writer_main.cc]: Should there be a TODO about handling sensors that are attached via non-fixed joints?

Comment (SirVer) [inline cartographer_ros/cartographer_ros/assets_writer_main.cc]: added further down.

**Pull request merged**